### DISCUSSION TITLE
Buttons from v17 to v18 changed heights and now they are bigger

### DISCUSSION THREAD

**Discussion by fexxdev** [IgniteUI/igniteui-angular#15055] [Q&A]
Hi everyone,
i've updated a project from v17 to v18, i've replaced every displayDensity token with the new --ig-size one, and now height 36px is missing. I have a production example of how in v17 the buttons height was in fact 36px, now if i set --ig-size  to medium i get 40px, if i set small then i get 32px.
Is there any way i can restore the previous behavior?

--- EDIT
im using $light-material-schema

**Comment by RivaIvanova** [comment]
Hi @fexxdev 

I have been looking into your question and after an investigation what I can say is that the values of the sizes are not changed, only the approach on how the size can be changed, i.e., removing the `displayDensity` property and using the `--ig-size` CSS property instead. Also, as mentioned in our documentation [here](https://www.infragistics.com/products/ignite-ui-angular/angular/components/general/update-guide#from-160x-to-161x), the new size-related variables correspond to the previous display density values as follows:
  - `compact` -> `--ig-size-small` -> `24px`
  - `cosy` -> `--ig-size-medium` -> `30px`
  - `comfortable` -> `--ig-size-large` -> `36px`

I have prepared two samples that demonstrate the button sizes in version 17.0.x and version 18.0.x. Please note that the `raised` button has been renamed to `contained` in version 17.1.0 as mentioned in our Changelog [here](https://github.com/IgniteUI/igniteui-angular/blob/master/CHANGELOG.md#general-6).
 - [Button Sizes version 18.0.x](https://stackblitz.com/edit/dn66yd-hh8ksc?file=src%2Fapp%2Fapp.component.scss,src%2Fapp%2Fapp.component.html)
<img width="396" alt="image" src="https://github.com/user-attachments/assets/ce402d37-d83f-45ce-83f6-843862686c3b">

 - [Button Sizes version 17.0.x](https://stackblitz.com/edit/uwnkxg-tn3nuf?file=src%2Fapp%2Fapp.component.scss,src%2Fapp%2Fapp.component.html)
<img width="398" alt="image" src="https://github.com/user-attachments/assets/a8593dfb-e9c4-43b5-9021-dfc5d8477317">

Having the abovementioned in mind, I cannot tell what the reason for the button having height 40px might be. Could you please provide a small sample that demonstrates the behavior on your side? This will be extremely helpful in further investigating this matter and providing you with a solution as soon as possible.

Looking forward to your reply.

**Comment by fexxdev** [comment]
I'm looking into it... It's so weird, we have overrides for button theme but im using the documentation to do that. Still, medium is 40px, as soon as i comment out the theme the button are correct sizes. But inside the theme theres NO mention of heights, min heights, paddings or anything like that.
As soon as i find the culprit i will update this thread
